File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/di/AppMainComponent.java
package com.wordpress.lonelytripblog.funwithflags.di;

import com.wordpress.lonelytripblog.funwithflags.FunWithFlagApp;
import com.wordpress.lonelytripblog.funwithflags.ui.MainActivity;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {ViewModelModule.class,
        AndroidInjectionModule.class,
        ActivityContributor.class,
        ViewModelsProvider.class})
public interface AppMainComponent {

    void injectApp(FunWithFlagApp app);

}

File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/di/AppMainComponent.kt
package com.wordpress.lonelytripblog.funwithflags.di

import com.wordpress.lonelytripblog.funwithflags.FunWithFlagsApp
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [FunWithFlagsModule::class, ActivityScopeModule::class,
    AndroidInjectionModule::class,
    ActivityContributor::class,
    ViewModelsProvider::class])
interface AppMainComponent {
    fun injectApp(app: FunWithFlagsApp)
}